mantidproject/mantid
# Interactive Tool use RangeMarker instead of VerticalMarker's

**Description of work.**
This PR replaces the VerticalMarkers used in the interactive_tool for Fitting plots with a RangeMarker. Do this has provided the following benefits:
- Dragging the markers when changing the fit range is far quicker (not as much lag)
- The RangeMarker cannot be dragged off the edge of the plot.
- The markers in the RangeMarker can be seamlessly switched. (i.e. the min and max markers can be moved either side of each other. The endx and startx will still be correct in the FitPropertyBrowser.)

**To test:**
1. Open the workbench
2. Load the data below
3. Right click and Plot spectrum 0.
4. Click `Fit`
5. Move the range markers around and check the correct values are displayed in the FitPropertyBrowser.
6. Choose a `Lorentzian` and `Flat Background`
7. Choose `Fit`->`Fit`
8. The fit should be produced between the range provided.

Compare this to the version in master, and it should have the benefits stated above.

**Data Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/3397207/irs26176_graphite002_red.zip)

Fixes #25898 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
